161018 1259 simulator m epics eemcu

tboegi opened the pull request:
The simulator, a driver and test cases "as is".
IOW: More cleanup needed.

Replaces
https://github.com/epics-modules/motor/pull/2

tboegi committed:
Add simulator

Add a simulator for a motion controller talking TCP/IP.
This is a replacement for the old simulator, mainly for
training and testing.
The simulator has 2 frontendd: the commands we use for thr
EtherCAT based systems, the IcePAP from ESRF and the 3rd one
is disabled right now.
Today the commands don't overlapp, and the simulator accepts both
labuages.

It may be wise to select only one protocol in the future.

tboegi committed:
Add m-epics-eemcu

Add an example of an IOC which can be used with the simulator.
The code is "as is".

tboegi committed:
Add nose test

Add the limited test suite that we have today, running against the EtherCAT
based test crate.
The python scripts need some refactoring.

tboegi committed:
startup/st.Sim2.cmd: snippets are no longer under snippets/

mp49 commented:
Is the simulator for the EtherCAT hardware? If so, I would have a deeper directory structure, to allow simulators of other hardware (I know that DLS wrote one for the Turbo PMAC). For example:

test/simulator/EtherCAT/controllerX/
test/simulator/EtherCAT/controllerY/
test/simulator/DeltaTau/GeobrickLV/
test/simulator/Newport/XPS/

etc.

tboegi committed:
Move the simulator code into a subdirectory

Move the code from TCPsimulator/ into the deeper directory structure
simulator/EtherCAT/ to allow more simulators to be added.

tboegi committed:
mv test_nose/ nose_tests

tboegi committed:
Add variuos runXXX.sh scripts

Add 3 shell scripts to start the simulator, the EPICS IOC and to run the test
suite

tboegi commented:
Good point, thanks.
We will hower try to simulate EtherCAT hardware in one simulator to minimize the
maintenance, so I moved it into
test/simulator/EtherCAT

tboegi committed:
Change port to 5000

tboegi commented:
Will be replaced by a new one
